src/lib/contracts.ts

```diff
@@ -198,67 +255,37 @@ export function validateContract(contract: unknown): boolean {
 
 	return true;
 }
+
 export function validateTargetContracts(
 	apps: TargetApps,
-): Dictionary<ApplicationContractResult> {
-	return Object.keys(apps)
-		.map((appUuid): [string, ApplicationContractResult] => {
-			const app = apps[appUuid];
-			const [release] = Object.values(app.releases);
-			const serviceContracts = Object.keys(release?.services ?? [])
-				.map((serviceName) => {
-					const service = release.services[serviceName];
-					const { contract } = service;
-					if (contract) {
-						try {
-							// Validate the contract syntax
-							validateContract(contract);
-
-							return {
-								serviceName,
-								contract,
-								optional: checkTruthy(
-									service.labels?.['io.balena.features.optional'],
-								),
-							};
-						} catch (e: any) {
-							throw new ContractValidationError(serviceName, e.message);
+): Dictionary<AppContractResult> {
+	return Object.fromEntries(
+		Object.entries(apps).flatMap(([appUuid, app]) =>
```

Comment:
I think this can just be a normal `map`:
```suggestion
		Object.entries(apps).map(([appUuid, app]) =>
```
which if so would be much better to make the intent clearer. I did try to spot if the flat map was actually required and it don't think it is, so that has either added confusion or this section is confusing enough about why a `flatMap` is required that I think clarifying it would be very beneficial

Reply:
You are right, I think an earlier version could return an empty array for some conditions on the map of `app.releases` which is why I used `flatMap`

Reply:
Actually, no, it used a `flatMap` because the mapping over releases also returns a pair. I'll use a `for` loop as you suggested as that makes it clearer